File: 01_Leetcode_FAANG/Swagat_Shekhar/Matrix_Diagonal_Sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int sum =0;
        for(int i=0;i<mat.length;i++){
            for(int j=0;j< mat[i].length;j++)
            {
                if(i==j)
                {
                    sum = sum + mat[i][j];
                }
                if((i+j)==mat.length-1 && i!=j)
                {
                    sum = sum + mat[i][j];
                }
            }
        }
        return sum;
    }
}

File: 01_Leetcode_FAANG/Rajanikant_Pattnaik/matrix_diagonal_sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int n=mat.size();
        int sum=0;
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
              if(i==j||n-1-i==j||n-1-j==i){
                  sum=sum+mat[i][j];
              }  
            }
        }
        return sum;
    }
};
